### PR TITLE
Support Java 1.20.2

### LINKED ISSUE #4150: Java 1.20.2 is not supported
### Describe the bug

Loghin with Bedrock is not possible, players are kicked with an outdated client.

### To Reproduce

Login on Server

### Expected behaviour

If a Bedrock player wants to log in, he gets the message that he is using an outdated client.

### Screenshots / Videos

image

### Server Version and Plugins

This server is running BungeeCord version git:BungeeCord-Bootstrap:1.20-R0.2-SNAPSHOT:fed646d:1748 by md_5

07:22:29 [INFO] BPM+ > Plugins (21 + 7): BetterSecurity, BungeeChat, BungeePluginManagerPlus, BungeeSpy, BungeeTokens, CensorReloaded, CMIB, CraftingStore, ExceptionAnnouncer, floodgate, Geyser-BungeeCord, LiteBans, LuckPerms, Maintenance, MarriageMaster, NoVPN, NuVotifier, RankGift, RankUpgrade, spark, UltimateAutoRestartPassthrough, cmd_alert, cmd_find, cmd_kick, cmd_list, cmd_send, cmd_server, reconnect_yaml.

I try it with and without Viaversion.

### Geyser Dump

https://dump.geysermc.org/wkGzu1x8UUQi5DFLoAVnOMQdKXwkeXa9

### Geyser Version

Geyser-BungeeCord-1494

### Minecraft: Bedrock Edition Device/Version

1.20.30

### Additional Context

_No response_

### PR DESCRIPTION
**Pull request by Konicai**
Closes #4150

### CODE REVIEW

**Comment by Kas-tle** [comment]
FYI to anyone wondering why this isn't merged, we will likely wait for resolution of this https://github.com/SpigotMC/BungeeCord/issues/3519 as currently even if this was merged, it would not be viable to run Geyser on most proxy setups without running Geyser Standalone in front of ViaProxy in front of Bungee/Waterfall. Though this should currently function OK for simple 1.20.2 Spigot server if people wish to use the attached build artifacts from the latest commit.

**Comment by GrahamSH-LLK** [comment]
> FYI to anyone wondering why this isn't merged, we will likely wait for resolution of this [SpigotMC/BungeeCord#3519](https://github.com/SpigotMC/BungeeCord/issues/3519) as currently even if this was merged, it would not be viable to run Geyser on most proxy setups without running Geyser Standalone in front of ViaProxy in front of Bungee/Waterfall. Though this should currently function OK for simple 1.20.2 Spigot servers if people wish to use the attached build artifacts from the latest commit.

That issue seems to be resolved.

**Comment by maisymoe** [comment]
Due to a few unfortunate circumstances with my server that I won't detail, I'm using this (and the floodgate) PR build in what is effectively production.

Everything appears to work flawlessly, but Bedrock players seem to be receiving entirely scrambled, often negative potion effects. I can only assume this is due to Mojang no longer using numeric effect IDs in certain scenarios.

A wither skeleton gave poison, a regen potion gave nausea, a haste beacon gave slowness.

I suppose this could also be a general Geyser issue unrelated to 1.20.2/this PR, but it only seems logical with the effect ID changes I mentioned earlier.

My server runs Paper 1.20.2 #201, with what I believe are the latest artifacts from this and the floodgate PR. I do have other plugins, so I'll spin up a second server and follow up if I can reproduce the issue.

**Comment by Kas-tle** [comment]
If possible testing on 1.20.1 to see if the issue persists would be appreciated.

**Comment by maisymoe** [comment]
> If possible testing on 1.20.1 to see if the issue persists would be appreciated.

Will do.

**Comment by maisymoe** [comment]
Hello again, I have confirmed that I can confidently reproduce this bug. My findings:

# Java 1.20.1
* Using the current stable Geyser build with the recently merged floodgate 1.20.2 build, potion effects behave as expected. 

* It is worth noting that upon installing the latest ViaVersion dev build, and using the build of Geyser _from this PR_ on 1.20.1, the issue can be reproduced.

# Java 1.20.2
* Using the build of Geyser from this PR and the aforementioned floodgate build, I can reproduce the potion effects being scrambled.

I have uploaded a demonstration video to my image host as it is too large for GitHub:
https://i.maisy.moe/4d6be396a.mp4

(I apologise for the Discord noise about halfway through...)

**Comment by maisymoe** [comment]
Also, somewhat unrelated, but both on my on server and during this test I noticed that Bedrock player skins are somewhat inconsistent in whether or not they appear.

**Comment by Konicai** [comment]
thank you

**Comment by GrahamSH-LLK** [comment]
I'm running this in production for various reasons. I've had no problems other than the ones mentioned by @maisymoe earlier. After reading over logs, there have been no errors since I installed this PR and my bedrock players have had no issues!

File: core/src/main/java/org/geysermc/geyser/translator/protocol/java/entity/player/JavaPlayerInfoUpdateTranslator.java [hunk 1]
```diff
@@ -37,10 +37,12 @@
 import org.geysermc.geyser.skin.SkinManager;
 import org.geysermc.geyser.translator.protocol.PacketTranslator;
 import org.geysermc.geyser.translator.protocol.Translator;
+import org.jetbrains.annotations.Nullable;
```
Comment (Kas-tle) [inline core/src/main/java/org/geysermc/geyser/translator/protocol/java/entity/player/JavaPlayerInfoUpdateTranslator.java:40]: Did you mean to use checker here?

Comment (Konicai) [reply core/src/main/java/org/geysermc/geyser/translator/protocol/java/entity/player/JavaPlayerInfoUpdateTranslator.java:40]: yeppers

**Review by Kas-tle — APPROVED** [review]

**Review by Camotoy — APPROVED** [review]

**Pull request merged**